src/Stripe.net/Infrastructure/JsonConverters/PaymentIntentSourceActionConverter.cs

```diff
@@ -22,24 +22,24 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var sourceAction = new PaymentIntentSourceAction();
-
             if (reader.TokenType == JsonToken.Null)
             {
-                sourceAction.Type = PaymentIntentSourceActionType.None;
-                return sourceAction;
+                return null;
             }
 
             var incoming = JObject.Load(reader);
+            var sourceAction = new PaymentIntentSourceAction()
+            {
+                Type = PaymentIntentSourceActionType.Unknown,
+                Value = default(IPaymentIntentSourceActionValue),
+            };
+
+            var type = incoming.SelectToken("type")?.ToString();
 
-            if (incoming.SelectToken("type")?.ToString() == "authorize_with_url")
+            if (type.Equals("authorize_with_url"))
             {
                 sourceAction.Type = PaymentIntentSourceActionType.AuthorizeWithUrl;
-                sourceAction.AuthorizeWithUrl = Mapper<PaymentIntentSourceActionAuthorizeWithUrl>.MapFromJson(incoming.SelectToken("value")?.ToString());
-            }
-            else
-            {
-                sourceAction.Type = PaymentIntentSourceActionType.Unknown;
+                sourceAction.Value = Mapper<PaymentIntentSourceActionAuthorizeWithUrl>.MapFromJson(incoming.SelectToken("value")?.ToString());
```

Comment:
OOC can't you do the same logic as the other one where you map based on `type` but use `value` for the deserialization? Or is it just that we have one example so it's cleaner this way for now?